### PR TITLE
chore: course dates feature

### PR DESCRIPTION
**Pull request by mumer92**
This is continuity of https://github.com/openedx/openedx-app-ios/pull/137
to address feedback, as original PR was merged.

This PR also fixes edge case for the Today block and siblings for Today block, small color fixes to Circle and bit of refactoring.

### CODE REVIEW

File: Course/Course/Presentation/Dates/CourseDatesView.swift [hunk 1]
```diff
@@ -35,6 +35,7 @@ public struct CourseDatesView: View {
                     }
                 } else if let courseDates = viewModel.courseDates, !courseDates.courseDateBlocks.isEmpty {
                     CourseDateListView(viewModel: viewModel, courseDates: courseDates)
+                        .padding(.top)
```
Comment (volodymyr-chekyrta) [inline Course/Course/Presentation/Dates/CourseDatesView.swift:38]: Please set a specific value for `.padding(.top)`. 
You might use `.padding(.top, 20)` as an example.

Comment (mumer92) [reply Course/Course/Presentation/Dates/CourseDatesView.swift]: @volodymyr-chekyrta i have added padding value to this, but i think this should be handled by the parent or maybe we should handle this by adding NavigationTtile to the child view. This same padding/overlapping issue will occur on the handouts tab.

File: Course/CourseTests/Presentation/Unit/CourseDateViewModelTests.swift [hunk 2]
```diff
@@ -5,12 +5,12 @@
 //  Created by Muhammad Umer on 10/24/23.
 //
 
-import SwiftyMocky
+import SwiftUI
```
Comment (saeedbashir) [inline Course/CourseTests/Presentation/Unit/CourseDateViewModelTests.swift:8]: Tests are running fine without this import `SwiftUI` .

**Review by saeedbashir — COMMENTED** [review]
Just a small comment otherwise working fine

**Review by saeedbashir — APPROVED** [review]
LGTM 👍

**Review by volodymyr-chekyrta — APPROVED** [review]
LGTM

**Pull request merged**